lib/gooddata/core/loggers/gd_logger.rb

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require 'logger'
+require_relative '../../rest/rest_aggregator'
+
+module GoodData
+  class GDLogger
+    severity_map = { debug: Logger::DEBUG, info: Logger::INFO, warn: Logger::WARN, error: Logger::ERROR, fatal: Logger::FATAL, unknown: Logger::UNKNOWN }
+
+    attr_accessor :execution_id, :loggers
+
+    def initialize
+      @loggers = {}
+    end
+
+    def logging_on(logger_id, logger)
+      loggers[logger_id] = logger
+    end
+
+    def logging_off(logger_id)
+      loggers[logger_id] = NilLogger.new
+    end
+
+    def logging_on?(logger_id)
+      loggers.key?(logger_id) && !loggers[logger_id].is_a?(NilLogger)
+    end
+
+    #
+    # Common logger methods
+    #
+
+    %i[debug error fatal info unknown warn].each do |severity|
+      define_method severity do |progname = nil, &block|
+        add(severity_map[severity], nil, progname, &block)
+      end
+    end
+
+    def add(severity, message, progname, modifiers: {}, filters: Set[], &block)
+      loggers.each do |pair|
+        logger_name = pair[0]
+        logger = pair[1]
+        next if filters.include? logger_name
+
+        if modifiers.key? logger_name
```

Comment:
info: Refers to 'modifiers' more than self (maybe move it to another class?) (FeatureEnvy) (reek)